tests/integration/ambient/registry_setup_test.go

```diff
@@ -35,11 +35,23 @@ const (
 )
 
 func testRegistrySetup(ctx resource.Context) (err error) {
-	registry, err = registryredirector.New(ctx, registryredirector.Config{
-		Cluster:        ctx.AllClusters().Default(),
-		TargetRegistry: "kind-registry:5000",
-		Scheme:         "http",
-	})
+	var config registryredirector.Config
+
+	isKind := ctx.Clusters().IsKindCluster()
+
+	// By default, for any platform, the test will pull the test image from public "gcr.io" registry.
+	// For "Kind" environment, it will pull images from the "kind-registry".
+	// "Kind" environment in IPv6 format, unable to properly resolve external images path, like "gcr.io"
```

Comment:
```suggestion
	// For kind clusters, it will pull images from the "kind-registry" local registry container. In IPv6-only clusters
	// this is not supported as CoreDNS will attempt to "talk" to the upstream DNS over docker/IPv4 for resolution.
	// For context: https://github.com/kubernetes-sigs/kind/issues/3114
```